read_csv with Ray engine with 'lineterminator' parameter outputs incorrect result

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **Modin version** : ce2bea8867739ea792cdda0dce8ad50e366ebae1
- **Python version**: 3.8.6
- **Code we can use to reproduce**: 
```
import os

os.environ["MODIN_ENGINE"] = "ray"

import pandas
import modin.pandas as pd
from modin.pandas.test.utils import df_equals

test_filename = "test.csv"
kwargs = {
    "filepath_or_buffer": test_filename,
    "lineterminator": "x",
}
str_two_cols = "col1,col2x1,2x3,4x"

try:
    with open(test_filename, "w") as f:
        f.write(str_two_cols)

    df_pandas = pandas.read_csv(**kwargs)
    print("pandas.read_csv output:\n", df_pandas)
    df_pd = pd.read_csv(**kwargs)
    print("pd.read_csv output:\n", df_pd)
    df_equals(df_pandas, df_pd)
finally:
    os.remove(test_filename)
```

<!--
You can obtain the Modin version with

python -c "import modin; print(modin.__version__)"
-->

### Describe the problem
<!-- Describe the problem clearly here. -->

### Source code / logs
<!-- Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem. -->
```
pandas.read_csv output:
    col1  col2
0     1     2
1     3     4
pd.read_csv output:
 Empty DataFrame
Columns: [col1, col2]
Index: []
Traceback (most recent call last):
  File "test.py", line 125, in <module>
    df_equals(df_pandas, df_pd)
  File "/modin/pandas/test/utils.py", line 511, in df_equals
    assert False, "One of the passed frames is empty, when other isn't"
AssertionError: One of the passed frames is empty, when other isn't
```

--- a/modin/core/io/text/text_file_dispatcher.py
+++ b/modin/core/io/text/text_file_dispatcher.py
@@ -655,6 +655,9 @@ class TextFileDispatcher(FileDispatcher):
         if read_kwargs["chunksize"] is not None:
             return (False, "`chunksize` parameter is not supported")
 
+        if read_kwargs["lineterminator"] is not None:
+            return (False, "`lineterminator` parameter is not supported")
+
         skiprows_supported = True
         if is_list_like(skiprows_md) and skiprows_md[0] < header_size:
             skiprows_supported = False

--- a/modin/pandas/test/test_io.py
+++ b/modin/pandas/test/test_io.py
@@ -715,9 +715,7 @@ class TestCsv:
         escapechar,
         dialect,
     ):
-        if Engine.get() != "Python" and lineterminator == "x":
-            pytest.xfail("read_csv with Ray engine outputs empty frame - issue #2493")
-        elif Engine.get() != "Python" and escapechar:
+        if Engine.get() != "Python" and escapechar:
             pytest.xfail(
                 "read_csv with Ray engine fails with some 'escapechar' parameters - issue #2494"
             )

